### ISSUE TITLE
wifi_internface_t has not been declared 

### ISSUE THREAD

**Issue by tgb-sw** [Pedroalbuquerque/ESPBASE#12]
Hi Pedro,

I'm new to platformio/esp8266. Trying to use ESPBASE. Unfortunately I'm not able to build this. Tried in an own project, and also with your sample I get:

```

In file included from C:/Daten/PlatformIO/Projects/espbase/ESPBASE/Examples/WiFi_CFG_OTA_Telnet/src/WiFi_CFG_OTA_Telnet.ino:3:0:
src\ESPBASE.h:78:46: error: 'wifi_interface_t' has not been declared
     void setWifiProtocol ( uint8_t protocol ,wifi_interface_t current_wifi_interface);
                                              ^
In file included from C:/Daten/PlatformIO/Projects/espbase/ESPBASE/Examples/WiFi_CFG_OTA_Telnet/src/WiFi_CFG_OTA_Telnet.ino:3:0:
src\ESPBASE.h: In member function 'bool ESPBASE::setWifiPower(float)':
src\ESPBASE.h:137:37: error: void value not ignored as it ought to be
     return WiFi.setOutputPower(power);
                                     ^
src\ESPBASE.h: At global scope:
src\ESPBASE.h:143:49: error: 'wifi_interface_t' has not been declared
 void ESPBASE::setWifiProtocol(uint8_t protocol, wifi_interface_t current_wifi_interface ){
                                                 ^
src\ESPBASE.h: In member function 'void ESPBASE::setWifiProtocol(uint8_t, int)':
src\ESPBASE.h:146:30: error: invalid conversion from 'uint8_t {aka unsigned char}' to 'WiFiPhyMode_t {aka WiFiPhyMode}' [-fpermissive]
     WiFi.setPhyMode( protocol); //set 802.11b mode to increase range
                              ^
In file included from C:\Users\Test\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src/ESP8266WiFiSTA.h:28:0,
                 from C:\Users\Test\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src/ESP8266WiFi.h:34,
                 from src\ESPBASE.h:47,
Compiling .pio\build\esp8266\lib8ce\ESP8266mDNS\ESP8266mDNS.cpp.o
                 from C:/Daten/PlatformIO/Projects/espbase/ESPBASE/Examples/WiFi_CFG_OTA_Telnet/src/WiFi_CFG_OTA_Telnet.ino:3:
C:\Users\Test\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src/ESP8266WiFiGeneric.h:77:14: error:   initializing argument 1 of 'bool ESP8266WiFiGenericClass::setPhyMode(WiFiPhyMode_t)' [-fpermissive]
         bool setPhyMode(WiFiPhyMode_t mode);
Compiling .pio\build\esp8266\lib8ce\ESP8266mDNS\ESP8266mDNS_Legacy.cpp.o
Compiling .pio\build\esp8266\lib8ce\ESP8266mDNS\LEAmDNS.cpp.o
              ^
Compiling .pio\build\esp8266\lib8ce\ESP8266mDNS\LEAmDNS_Control.cpp.o
Compiling .pio\build\esp8266\lib8ce\ESP8266mDNS\LEAmDNS_Helpers.cpp.o
In file included from C:/Daten/PlatformIO/Projects/espbase/ESPBASE/Examples/WiFi_CFG_OTA_Telnet/src/WiFi_CFG_OTA_Telnet.ino:3:0:
src\ESPBASE.h: In member function 'bool ESPBASE::setWifiPower(float)':
src\ESPBASE.h:141:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
*** [.pio\build\esp8266\src\WiFi_CFG_OTA_Telnet.ino.cpp.o] Error 1

```
I use vscode 1.44.1 platformio core 4.3.1 Home 3.1.1

I guess, I do something wrong :-/ Maybe I'm missing additionals libs or includes? 
If you have an idea what to check, I will do. If you guess it must be some sort of beginner problem, don't mind and just close the issue.

Best regards

Thomas

**Comment by Pedroalbuquerque** [comment]
HI Thomas

thank you for your interest in my lib. I am very sorry for taking so long to reply.

I have missed to test this new code in an ESP8266.
I have done the improvements as I am working on ESP32 project.
Realising that this project requirements are common to all project and were not available , I merged the branch I was working with, to the master branch.
I missed to test it on esp8266,reason why you are now facing this error.

I am (as I write) working to fix this and I expect to have this solved by now .

Please try again

**Comment by tgb-sw** [comment]
Hi Pedro 

thank you very much for your work. After updating, I'm able to build and upload and the lib now seems to work as expected.

Unfortunately I'm not able to do ota updates. It uploads fine, but than the boot fails:
```
Uploading: [=====================================================       ] 89%
Uploading: [======================================================      ] 89%
Uploading: [======================================================      ] 89% 
Uploading: [======================================================      ] 90%
Uploading: [=======================================================     ] 90%
Uploading: [=======================================================     ] 91% 
Uploading: [=======================================================     ] 91% 
Uploading: [=======================================================     ] 92%
Uploading: [========================================================    ] 92% 
Uploading: [========================================================    ] 93%
Uploading: [========================================================    ] 93% 
Uploading: [========================================================    ] 94% 
Uploading: [=========================================================   ] 94%
Uploading: [=========================================================   ] 95% 
Uploading: [=========================================================   ] 95%
Uploading: [==========================================================  ] 95% 
Uploading: [==========================================================  ] 96% 
Uploading: [==========================================================  ] 96%
Uploading: [==========================================================  ] 97%
Uploading: [=========================================================== ] 97% 
Uploading: [=========================================================== ] 98% 
Uploading: [=========================================================== ] 98%
Uploading: [============================================================] 99% 
Uploading: [============================================================] 99%
Uploading: [============================================================] 100% Done...


12:29:21 [INFO]: Waiting for result...
12:29:22 [INFO]: Result: OK
=================================================================================================================================== [SUCCESS] Took 7.98 seconds ===================================================================================================================================
--- Available filters and text transformations: colorize, debug, default, direct, esp8266_exception_decoder, hexlify, log2file, nocontrol, printable, send_on_enter, time
--- More details at http://bit.ly/pio-monitor-filters
--- Miniterm on COM4  115200,8,N,1 ---
--- Quit: Ctrl+C | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H ---

End

 ets Jan  8 2013,rst cause:4, boot mode:(3,6)

wdt reset
load 0x4010f000, len 3656, room 16
tail 8
chksum 0x0c
csum 0x0c
v9c56ed1f
@cp:0
ld
e:
 ets Jan  8 2013,rst cause:3, boot mode:(3,6)

ets_main.c
```
I first thought it's another problem with the etsbase lib. Then I reverted back to my code without etsbase - and encountered there the same error after an ota update :-/ Serial uploads still work with platformio.

In my own code, ota worked fine before, but even after reducing to the standard ota sample code, I still get this error. If I Download the same code with arduino ide, over ota everything works without the  above error. If I try platformio again, i get the error again.

I think it has something to do with platformio, I don't know why it worked for several builds and ota updates before. But seems to have nothing to do with your lib.

Your code now compiles and works, I think we can close this issue.

Best regards,

Thomas

**Comment by tgb-sw** [comment]
Hi Pedro,

my problem "no boot after ota" was in the framework see https://github.com/esp8266/Arduino/issues/7267 and has been fixed with 2.7.1.

Best regards,

Thomas

**Comment by Pedroalbuquerque** [comment]
Hi Thomas,

great that it worked in the end :) thank you for sharing the pitfall
